lib/grant.js

```diff
@@ -140,6 +141,9 @@ function checkClient (done) {
       return done(error('invalid_client', 'Client credentials are invalid'));
     }
 
+    self.client.accessTokenLifetime = client.accessTokenLifetime;
+    self.client.refreshTokenLifetime = client.refreshTokenLifetime;
```

Comment:
I avoided this here because in the extractCredentials (lines 83-90), this.client is defined 

```
this.client = credsFromBasic(this.req) || credsFromBody(this.req);
```

So I am just adding these values to the client object. Alternatively, I could do something like 
```
self.accessTokenLifetime = client.accessTokenLifetime;
self.refreshTokenLifetime = client.refreshTokenLifetime;
```

Or, alternatively, I could refactor and make the values assigned to client in lines 83-90 into some different variable like "credentials" or "clientCredentials"